kubernetes/samples/scripts/domain-lifecycle/restartServer.sh

```diff
@@ -0,0 +1,103 @@
+#
+
+script="${BASH_SOURCE[0]}"
+scriptDir="$( cd "$( dirname "${script}" )" && pwd )"
+source ${scriptDir}/helper.sh
+if [ "${debug}" == "true" ]; then set -x; fi;
+set -eu
+
+function usage() {
+
+  cat << EOF
+
+  This script restarts a running WebLogic server in a domain by deleting the server pod.
+ 
+  Usage:
+ 
+    $(basename $0) -s myserver [-n mynamespace] [-d mydomainuid] [-m kubecli]
+  
+    -s <server_name>           : Server name parameter is required.
+
+    -d <domain_uid>            : Domain unique-id. Default is 'sample-domain1'.
+
+    -n <namespace>             : Domain namespace. Default is 'sample-domain1-ns'.
+    
+    -m <kubernetes_cli>        : Kubernetes command line interface. Default is 'kubectl' 
+                                 if KUBERNETES_CLI env variable is not set. Otherwise the
+                                 default is the value of KUBERNETES_CLI env variable.
+
+    -h                         : This help.
+   
+EOF
+exit $1
+}
+
+kubernetesCli=${KUBERNETES_CLI:-kubectl}
+serverName=""
+clusterName=""
+domainUid="sample-domain1"
+domainNamespace="sample-domain1-ns"
+
+while getopts "vs:m:n:d:h" opt; do
+  case $opt in
+    s) serverName="${OPTARG}"
+    ;;
+    n) domainNamespace="${OPTARG}"
+    ;;
+    m) kubernetesCli="${OPTARG}"
+    ;;
+    d) domainUid="${OPTARG}"
+    ;;
+    v) verboseMode=true;
+    ;;
+    h) usage 0
+    ;;
+    *) usage 1
+    ;;
+  esac
+done
+
+#
+# Function to perform validations, read files and initialize workspace
+#
+function initialize {
+
+  validateErrors=false
+
+  validateKubernetesCliAvailable
+  validateJqAvailable
+
+  # Validate that server name parameter is specified.
+  if [ -z "${serverName}" ]; then
+    validationError "Please specify the server name using '-s' parameter e.g. '-s managed-server1'."
+  fi
+
+  failIfValidationErrors
+}
+
+initialize
+
+# Get the domain in json format
+domainJson=$(${kubernetesCli} get domain ${domainUid} -n ${domainNamespace} -o json --ignore-not-found)
+if [ -z "${domainJson}" ]; then
+  printError "Unable to get domain resource for domain '${domainUid}' in namespace '${domainNamespace}'. Please make sure the 'domain_uid' and 'namespace' specified by the '-d' and '-n' arguments are correct. Exiting."
+  exit 1
+fi
+
+# Validate that specified server is either part of a cluster or is an independent managed server
+validateServerAndFindCluster "${domainUid}" "${domainNamespace}" "${serverName}" isValidServer clusterName isAdminServer
+if [ "${isValidServer}" != 'true' ]; then
+  printError "Server ${serverName} is not part of any cluster and it's not an independent managed server. Please make sure that server name specified is correct."
+  exit 1
+fi
+
+printInfo "Initiating restart of '${serverName}' by deleting server pod '${domainUid}-${serverName}'."
+result=$(${kubernetesCli} -n ${domainNamespace} delete pod ${domainUid}-${serverName} --ignore-not-found)
```

Comment:
If this command fails, what happens?  I suspect it's possible that it can put something in 'stdout', in which case thsi script would incorrectly print "Server restart succeeded".

Reply:
I have `set -eu` in the beginning so the script should abort if this command fails. I tried temporarily setting `kubernetesCli` value to "fake_kubectl" (before this command) to make it fail and the script failed with the below error. I can check the command return value if you think that's necessary, I thought it was not needed.

$ restartServer.sh -s managed-server3 -d domain1 -n weblogic-domain-1
[2021-03-25T02:02:11.430 UTC][INFO] Initiating restart of 'managed-server3' by deleting server pod 'domain1-managed-server3'.
restartServer.sh: line 98: fake_kubectl: command not found

Reply:
I forgot about teh 'set -eu'.